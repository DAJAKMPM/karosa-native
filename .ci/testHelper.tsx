import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
