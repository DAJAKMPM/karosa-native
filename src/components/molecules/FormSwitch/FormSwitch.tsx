/**
 *
 * FormSwitch
 * @format
 *
 */

import React, { FC, useState, useEffect, useCallback } from "react";
import { useField } from "formik";
import Switch from "@app/atoms/Switch";

import type { PropsType } from "./types";

const FormSwitch: FC<PropsType> = (props) => {
  const { name, disabled } = props;

  const [, meta, helpers] = useField(name);

  const [currentValue, setCurrentValue] = useState<boolean>(
    meta.value || meta.initialValue
  );

  useEffect(() => {
    helpers.setValue(currentValue);
  }, [currentValue]);

  const handleSwitch = useCallback(
    (value: boolean) => {
      setCurrentValue(value);
      helpers.setTouched(true);
    },
    [helpers]
  );

  return (
    <Switch onChange={handleSwitch} value={currentValue} disabled={disabled} />
  );
};

export default FormSwitch;
