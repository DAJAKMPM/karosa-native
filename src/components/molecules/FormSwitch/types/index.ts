/**
 *
 * @format
 *
 */

export type PropsType = {
  name: string;
  disabled?: boolean;
};
