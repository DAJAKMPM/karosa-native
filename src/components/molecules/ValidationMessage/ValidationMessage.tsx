/**
 *
 * ValidationMessage
 * @format
 *
 */

import React, { FC } from "react";
import { ErrorMessage } from "formik";
import Text from "@app/atoms/Text";

import type { PropsType } from "./types";
import ValidationMessageStyles from "./styles";

const ValidationMessage: FC<PropsType> = (props) => {
  const { name } = props;

  return (
    <ErrorMessage name={name}>
      {(msg) => <Text customStyle={ValidationMessageStyles.text} text={msg} />}
    </ErrorMessage>
  );
};

export default ValidationMessage;
