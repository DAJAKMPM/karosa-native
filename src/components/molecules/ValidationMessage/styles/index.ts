/**
 *
 * @format
 *
 */

import {StyleSheet} from 'react-native';

const ValidationMessageStyles = StyleSheet.create({
});

export default ValidationMessageStyles;
