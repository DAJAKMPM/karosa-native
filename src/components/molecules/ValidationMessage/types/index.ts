/**
 *
 * @format
 *
 */
import { ErrorMessageProps } from "formik";

export type PropsType = ErrorMessageProps;
