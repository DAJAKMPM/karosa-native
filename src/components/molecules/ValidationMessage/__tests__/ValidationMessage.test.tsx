import React from "react";
import { renderWithFormikWrapper } from "@app/ci/testHelper";
import FormSwitch from "@app/molecules/FormSwitch";
import ValidationMessage from "../ValidationMessage";

describe("ValidationMessage Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithFormikWrapper(
      { test: false },
      <>
        <FormSwitch name="test" />
        <ValidationMessage name="test" />
      </>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
