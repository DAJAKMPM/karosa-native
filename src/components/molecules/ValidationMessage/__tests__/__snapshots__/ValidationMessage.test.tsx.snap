// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationMessage Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="#0AA351"
    style={
      Object {
        "height": 31,
        "width": 51,
      }
    }
    tintColor="#BDBDBD"
    value={false}
  />
</View>
`;
